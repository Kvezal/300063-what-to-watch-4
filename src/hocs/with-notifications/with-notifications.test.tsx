import * as render from "react-test-renderer";
import * as React from "react";
import {Provider} from "react-redux";
import configureStore from 'redux-mock-store';

import ENameSpace from "@store/name-space";
import withNotifications from "@hocs/with-notifications/with-notifications";


const mockStore = configureStore([]);

const store = mockStore({
  [ENameSpace.NOTIFICATION]: {
    notifications: [],
  },
});

const TestComponent = () => {
  return <div className="test-component"/>;
};

const TestComponentWithHOC = withNotifications(TestComponent);

test(`should render component with NotificationList`, () => {
  const tree = render.create(
      <Provider store={store}>
        <TestComponentWithHOC/>
      </Provider>
  )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
