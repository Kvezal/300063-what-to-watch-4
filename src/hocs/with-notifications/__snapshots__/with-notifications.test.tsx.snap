// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withNotificationsHOC should render component with NotificationList 1`] = `
Array [
  <div
    className="test-component"
  />,
  <ul
    className="notification-list"
  />,
]
`;
